.changeset/yellow-moles-hammer.md

```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': minor
+---
+
+`SignatureChecker`: refactor `isValidSignatureNow` to make it safe if EIP-7377 (or similar) is ever implemented.
```

Comment:
I think the changelog should reflect the change rather than the rationale. This change looks more like:

```suggestion
`SignatureChecker`: refactor `isValidSignatureNow` to avoid validating smart contract signatures if the signer has code.
```

Reply:
I'd say "avoid validating ECDSA signatures" instead of "avoid validating smart contract signatures".

I think "smart contract signatures" can be confusing. Some people think of 1271 as smart contract signature.

Reply:
> I'd say "avoid validating ECDSA signatures" instead of "avoid validating smart contract signatures".

Yeah, I agree with that concern. Let's go with your first proposal.